albumentations/augmentations/transforms.py

```diff
@@ -3405,3 +3406,65 @@ def get_transform_init_args_names(self):
             "mask_fill_value",
             "random_offset",
         )
+
+
+class FDA(ImageOnlyTransform):
+    """
+    Fourier Domain Adaptation from https://github.com/YanchaoYang/FDA
+    Simple "style transfer".
+    Important: you need to pass target image as a parameter `target_image` in __call__, see example
+
+    Args:
+        beta_limit (float or tuple of float): coefficient beta from paper. Recommended less 0.3.
+
+    Targets:
+        image
+
+    Image types:
+        3-channel uint8 images only
+
+    Reference:
+        https://github.com/YanchaoYang/FDA
+        https://openaccess.thecvf.com/content_CVPR_2020/papers/Yang_FDA_Fourier_Domain_Adaptation_for_Semantic_Segmentation_CVPR_2020_paper.pdf
+
+    Example:
+        >>> import numpy as np
+        >>> import albumentations as A
+        >>> image = np.random.randint(0, 256, [100, 100, 3], dtype=np.uint8)
+        >>> target_image = np.random.randint(0, 256, [100, 100, 3], dtype=np.uint8)
+        >>> aug = A.Compose([A.FDA(p=1)])
+        >>> result = aug(image=image, target_image=target_image)
+
+    """
+
+    def __init__(self, beta_limit=0.1, always_apply=False, p=0.5):
+        super(FDA, self).__init__(always_apply=always_apply, p=p)
+        self.beta_limit = to_tuple(beta_limit, low=0)
+
+    def apply(self, img, target_image=None, beta=0.1, **params):
+        if target_image is None:
+            return img
+        return F.fourier_domain_adaptation(img=img, target_img=target_image, beta=beta)
+
+    def get_params_dependent_on_targets(self, params):
+        img = params["image"]
+        target_img = params["target_image"]
+        target_img = cv2.resize(target_img, img.shape[1::-1])
+
+        if target_img.shape[2] != img.shape[2]:
+            raise ValueError(
+                "The source and target images must contain the same number of channels,"
+                f" but given {img.shape[2]} and {target_img.shape[2]} respectively."
+            )
```

Comment:
I think better to check that image contains third dimension:
```python
if target_img.ndim != img.ndim or (target_img.ndim == 3 and target_img.shape[2] != img.shape[2]):
            raise ValueError(
                "The source and target images must contain the same number of channels,"
                f" but given {img.shape[2]} and {target_img.shape[2]} respectively."
            )
```

Reply:
done.

Reply:
I will check all dimensions `target_img.shape != img.shape`. Ok?

Reply:
Oh yes, that's more correct!